processing/src/main/java/org/apache/druid/query/timeseries/TimeseriesQueryQueryToolChest.java

```diff
@@ -274,6 +274,24 @@ public boolean isCacheable(TimeseriesQuery query, boolean willMergeRunners)
             .build();
       }
 
+      @Override
+      public byte[] computeResultLevelCacheKey(TimeseriesQuery query)
+      {
+        final CacheKeyBuilder builder = new CacheKeyBuilder(TIMESERIES_QUERY)
+            .appendBoolean(query.isDescending())
+            .appendBoolean(query.isSkipEmptyBuckets())
+            .appendCacheable(query.getGranularity())
+            .appendCacheable(query.getDimensionsFilter())
+            .appendCacheables(query.getAggregatorSpecs())
+            .appendCacheable(query.getVirtualColumns())
+            .appendCacheables(query.getPostAggregatorSpecs())
+            .appendInt(query.getLimit());
+        if (query.isGrandTotal()) {
```

Comment:
This might as well just be `.appendBoolean(query.isGrandTotal())`, and is probably better, since there's less of a need to make sure that things we add _after_ it will mesh well.